File: codingTest/롤케이크자르기.py
from collections import Counter

def solution(topping):
    dic=Counter(topping)
    set_dic=set()
    answer=0
    for i in topping:
        dic[i]-=1
        set_dic.add(i)
        if dic[i]==0:
            dic.pop(i)
        if len(dic)==len(set_dic):
            answer+=1
    
    return answer

File: codingTest/simulation/롤케이크 자르기.js
function solution(topping) {
    let answer = 0;
    
    let eat_topping = {};
    for (let i = 0; i < topping.length; i++) {
        eat_topping[topping[i]] = (eat_topping[topping[i]] || 0) + 1;
    }
    
    const Cheolsu = new Set();
    let eat_cheolsu_brother = Object.keys(eat_topping).length;
    for (let i = 0; i < topping.length; i++) {
        Cheolsu.add(topping[i]);
        eat_topping[topping[i]] -= 1;
        
        if (eat_topping[topping[i]] === 0) {
            eat_cheolsu_brother -= 1;
        }
        
        if (Cheolsu.size === eat_cheolsu_brother) answer += 1;
    }
    
    return answer;
}